stdlib/tools/swift-reflection-test/swift-reflection-test.c

```diff
@@ -52,9 +52,18 @@
 #define NORETURN
 #endif
 
+typedef struct PipeMemoryReaderPage {
+  struct PipeMemoryReaderPage *Next;
+  swift_addr_t BaseAddress;
+  uint64_t Size;
+  char *Data;
+} PipeMemoryReaderPage;
+
+
 typedef struct PipeMemoryReader {
   int to_child[2];
   int from_child[2];
+  PipeMemoryReaderPage *Pages;
```

Comment:
I've added a simple linked list of "pages" to the PipeMemoryReader.  This works surprisingly well:  The RemoteMirror logic tends to make a lot of requests to the same area of memory, and this puts the most recent page at the top of the stack where it's found very quickly.